streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionJoinTest.java

```diff
@@ -354,9 +356,19 @@ private void createTopics() throws InterruptedException {
     }
 
 
-    private void startStreams() {
+    private void startStreams() throws InterruptedException {
         kafkaStreams = new KafkaStreams(builder.build(), streamsConfiguration);
+        final CountDownLatch startupLatch = new CountDownLatch(1);
+        kafkaStreams.setStateListener(new KafkaStreams.StateListener() {
+            @Override
+            public void onChange(final KafkaStreams.State newState, final KafkaStreams.State oldState) {
+                if (newState == KafkaStreams.State.RUNNING && oldState == KafkaStreams.State.REBALANCING) {
```

Comment:
Since we have three threads for this test, there can be multiple rebalances before the streams instance stabilize and start processing...